javascript/extractor/src/com/semmle/js/parser/TypeScriptASTConverter.java

```diff
@@ -1890,6 +1896,19 @@ private ITypeExpression convertThisParameterType(JsonObject function) throws Par
     return result;
   }
 
+  private IntList getOptionalParameterIndices(JsonObject function) throws ParseError {
+    IntList list = IntList.create(0);
+    int index = -1;
+    for (JsonElement param : getProperParameters(function)) {
+      ++index;
+      if (param.getAsJsonObject().has("questionToken")) {
+        list.add(index);
+        ;
```

Comment:
The line contains just a `;`. 

Also, I would get rid of the `++` and `-1` in this method. 
Maybe something like (untested): 
```Java
List<JsonElement> parameters = getProperParameters(function);
for (int i = 0; i < parameters.length; i++) {
  if (param.getAsJsonObject().has("questionToken")) {
    list.add(i);
  }
}
```

I know there is overhead in creating an additional list, but its more readable.

Reply:
Oh right, the dangers of not checking the output after auto-formatting 😅 

> I know there is overhead in creating an additional list, but its more readable.

Well I guess we just see things very differently here. I don't really want `getProperParameters` to build a copy every time it is called.